tests/scan/test_rewriting.py

```diff
@@ -1268,8 +1270,8 @@ def f_rnn_cmpl(u1_t, u2_t, x_tm1, y_tm1, y_tm3, W_in1):
 
         (pytensor_dump, pytensor_x, pytensor_y) = f4(v_u1, v_u2, v_x0, v_y0, vW_in1)
 
-        utt.assert_allclose(pytensor_x, v_x[-1:])
-        utt.assert_allclose(pytensor_y, v_y[-1:])
+        np.testing.assert_allclose(pytensor_x, v_x[-1:][0])
```

Comment:
Why the index, was it testing something with a different shape?

Reply:
```utt.assert_allclose()``` is able to extract the answer as it involves converting the two variables into numpy array and uses ```_allclose``` under the hood. Directly uses ```np.testing.assert_allclose``` does not allow us to compare ```pytensor_x``` and ```v_x[-1:][0]``` by directly writing np.testing.assert_allclose(pytensor_x, v_x[-1:]), which leads to assertion errors.

Reply:
I still don't understand it. My best guess it is a shape question and v_x[-1:] has one more degenerate dim than `pytensor_x`, in which case you should use `squeeze(0)` which is more readable / restrictive than [0]

Reply:
Yes was a shape mismacth thing. Squeeze solves it correctly.